src/script/lua_api/l_mainmenu.cpp

```diff
@@ -643,17 +643,33 @@ int ModApiMainMenu::l_create_world(lua_State *L)
 	const char *name	= luaL_checkstring(L, 1);
 	int gameidx			= luaL_checkinteger(L,2) -1;
 
+	std::string new_dir = fs::SanitizeFilename(name);
 	std::string path = porting::path_user + DIR_DELIM
-			"worlds" + DIR_DELIM
-			+ name;
+			+ "worlds" + DIR_DELIM + new_dir;
+
+	if (fs::PathExists(path)) {
+		path = path + "_";
+		for (int i = 1; i <= 100; i++) {
+			std::ostringstream test_path;
```

Comment:
No need for a full-blown `stringstream`, just use `std::to_string` and concatenation.